WordPress/Classes/ViewRelated/Site Creation/SiteInfo/SiteInformationWizardContent.swift

```diff
@@ -0,0 +1,159 @@
+        table.register(
+            InlineEditableNameValueCell.defaultNib,
+            forCellReuseIdentifier: InlineEditableNameValueCell.defaultReuseID
+        )
+    }
+
+    private func setupNextButton() {
+        nextStep.addTarget(self, action: #selector(goNext), for: .touchUpInside)
+
+        let buttonTitle = NSLocalizedString("Next", comment: "Button to progress to the next step")
+        nextStep.setTitle(buttonTitle, for: .normal)
+        nextStep.accessibilityLabel = buttonTitle
+        nextStep.accessibilityHint = NSLocalizedString("Navigates to the next step", comment: "Site creation. Navigates tot he next step")
+    }
+
+    private func setupHeader() {
+        let header = TitleSubtitleHeader(frame: .zero)
+        header.setTitle(headerData.title)
+        header.setSubtitle(headerData.subtitle)
+
+        table.tableHeaderView = header
+
+        NSLayoutConstraint.activate([
+            header.centerXAnchor.constraint(equalTo: table.centerXAnchor),
+            header.widthAnchor.constraint(equalTo: table.layoutMarginsGuide.widthAnchor),
+            header.topAnchor.constraint(equalTo: table.layoutMarginsGuide.topAnchor)
+            ])
+
+        table.tableHeaderView?.layoutIfNeeded()
+        table.tableHeaderView = table.tableHeaderView
+    }
+
+    @objc
+    private func goNext() {
+        guard let titleCell = cell(at: IndexPath(row: Rows.title.rawValue, section: 0)),
+            let taglineCell = cell(at: IndexPath(row: Rows.tagline.rawValue, section: 0)) else {
+            return
+        }
+
+        let collectedData = SiteInformation(title: titleCell.valueTextField.text ?? "", tagLine: taglineCell.valueTextField.text)
+        completion(collectedData)
+    }
+
+    private func cell(at: IndexPath) -> InlineEditableNameValueCell? {
+        return table.cellForRow(at: at) as? InlineEditableNameValueCell
+    }
+}
+
+extension SiteInformationWizardContent: UITableViewDataSource {
+    private enum TableStrings {
+        static let site = NSLocalizedString("Site Title", comment: "Site info. Title")
+        static let tagline = NSLocalizedString("Tagline", comment: "Site info. Tagline")
+        static let taglinePlaceholder = NSLocalizedString("Optional Tagline", comment: "Site info. Tagline placeholder")
+        static let footer = NSLocalizedString("The tagline is a short line of text shown right below the title", comment: "Site info. Table footer.")
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return Rows.count()
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = tableView.dequeueReusableCell(withIdentifier: InlineEditableNameValueCell.defaultReuseID, for: indexPath) as? InlineEditableNameValueCell {
+            configure(cell, index: indexPath)
+
+            return cell
+        }
+
+        return UITableViewCell()
```

Comment:
It strikes me that it might be preferable to use `assertionFailure` here rather than vending a stock table view cell. Thoughts?

Reply:
🤔 Sure!